crates/evm/evm/src/inspectors/stack.rs

```diff
@@ -232,15 +241,15 @@ macro_rules! call_inspectors {
             }
         )+
     };
-    (#[ret] [$($inspector:expr),+ $(,)?], |$id:ident $(,)?| $call:expr $(,)?) => {
+    (#[ret] [$($inspector:expr),+ $(,)?], |$id:ident $(,)?| $call:expr $(,)?) => {{
         $(
             if let Some($id) = $inspector {
                 if let Some(result) = ({ #[inline(always)] #[cold] || $call })() {
                     return result;
                 }
             }
         )+
-    };
+    }};
```

Comment:
Note the added inner scope grouping the inspectors making it so that an early return anywhere prevents the next ones from running